cli/azd/cmd/root.go

```diff
@@ -113,6 +113,10 @@ For more information, visit the Azure Developer CLI Dev Hub: https://aka.ms/azur
 	cmd.AddCommand(BuildCmd(opts, provisionCmdDesign, initInfraCreateAction, nil))
 	cmd.AddCommand(BuildCmd(opts, deployCmdDesign, initDeployAction, nil))
 
+	actions.Use(initDebugMiddleware())
+	actions.Use(middleware.UseTelemetry())
+	actions.Use(initCommandHooksMiddleware(opts))
```

Comment:
Is it possible to avoid the package-level variables? Could we move this into `RunE` -- which is really the `main` for the command?

Reply:
Middleware registration has been moved into the `RunE` function since it was easier to leverage some of the core top level configuration like `console` or `flags`.

Registered middlewares are still added to a package variable, but I don't see any issues with this at the moment.